.github/workflows/install-testing.yml

```diff
@@ -104,6 +105,7 @@ jobs:
     permissions:
       contents: read
     runs-on: ${{ matrix.os }}
+    if: ${{ github.repository == 'WordPress/wordpress-develop' || github.event_name == 'pull_request' }}
```

Comment:
If the second job depends on the first, then I'm not sure we need to also include this here. The second will never run if the condition is not met for `build-matrix`. Are there any reasons to explicitly list it here?

Reply:
By that same logic, that condition can be removed in quite a few other places throughout the workflows. I'd advocate for either consistently having it everywhere or consistently only having the conditions when needed and not handled via the `needs:` key.

Either is fine by me, but consistency is key IMO.